presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -576,7 +576,7 @@ private PlanWithProperties planTableScan(TableScanNode node, Expression predicat
                     deterministicPredicate,
                     types);
 
-            TupleDomain<ColumnHandle> simplifiedConstraint = decomposedPredicate.getTupleDomain()
```

Comment:
👍 
The same is used also in `PickTableLayout` and `IndexJoinOptimizer`. It would be nice to fix them too.

Reply:
I'm going to leave those alone for now. Especially, IndexJoinOptimizer's use of resolveIndex() has subtly different mechanics than what we're doing here with getTableLayouts().